explorer/ast/member.h

```diff
@@ -33,16 +43,20 @@ class Member {
  public:
   explicit Member(Nonnull<const Declaration*> declaration);
   explicit Member(Nonnull<const NamedValue*> struct_member);
+  explicit Member(Nonnull<const IndexedValue*> tuple_member);
 
   // The name of the member.
   auto name() const -> std::string_view;
+  // The index of the member, if any
+  auto index() const -> std::optional<size_t>;
```

Comment:
I think we should say that this method requires that `*this` represents a positional member rather than a named member, have it `CHECK`-fail internally if that's not the case, and otherwise return a `size_t`. That way the caller doesn't have to deal with unwrapping the `optional`.

Reply:
Done. Provided a couple of getters: `hasIndex()` and `hasName()` to discriminate between in places where they are used.